tribe-ext-settings-import-export.php

```diff
@@ -0,0 +1,370 @@
+                                    <input type="text" name="import_reset_confirmation"/><br/>
+                                    <?php _e( 'Enter "reset" into the above field if you would like to reset the settings.', 'PLUGIN_TEXT_DOMAIN' ); ?>
+                                </p>
+                                <p>
+                                    <input type="hidden" name="tribe_sie_action" value="reset_settings" />
+									<?php wp_nonce_field( 'tribe_sie_import_nonce', 'tribe_sie_import_nonce' ); ?>
+									<?php submit_button( __( 'Reset', 'PLUGIN_TEXT_DOMAIN' ), 'secondary', 'submit', false ); ?>
+                                </p>
+                            </form>
+                        </div><!-- .inside -->
+                    </div><!-- .postbox -->
+				</div><!-- .metabox-holder -->
+
+			</div><!--end .wrap-->
+
+			<?php
+		}
+
+		/**
+		 * Process a settings export that generates a .json file of the shop settings
+		 */
+		function tribe_sie_process_settings_export() {
+		    // Bail if no action
+			if( empty( $_POST['tribe_sie_action'] ) || 'export_settings' != $_POST['tribe_sie_action'] ) {
+				return;
+			}
+			// Bail if no nonce
+			if( ! wp_verify_nonce( $_POST['tribe_sie_export_nonce'], 'tribe_sie_export_nonce' ) ) {
+				return;
+			}
+			// Bail if no capability
+			if( ! current_user_can( 'manage_options' ) ) {
+				return;
+			}
+
+			$settings = get_option( 'tribe_events_calendar_options' );
+			ignore_user_abort( true );
+			nocache_headers();
+			header( 'Content-Type: application/json; charset=utf-8' );
+			header( 'Content-Disposition: attachment; filename=tribe-settings-export-' . date( 'm-d-Y' ) . '.json' );
+			header( "Expires: 0" );
+			echo json_encode( $settings );
+			exit;
+		}
+
+		/**
+		 * Process a settings import from a json file
+		 */
+		function tribe_sie_process_settings_import() {
+
+		    // Bail if no action
+			if( empty( $_POST['tribe_sie_action'] ) || 'import_settings' != $_POST['tribe_sie_action'] ) {
+				return;
+			}
+
+			// Bail if no nonce
+			if( ! wp_verify_nonce( $_POST['tribe_sie_import_nonce'], 'tribe_sie_import_nonce' ) ) {
+				return;
+			}
+
+			// Bail if no capability
+			if( ! current_user_can( 'manage_options' ) ) {
+				return;
+			}
+
+			$import_file = $_FILES['import_file']['tmp_name'];
+			if( empty( $import_file ) ) {
+				wp_die( __( 'Please upload a file to import.', 'PLUGIN_TEXT_DOMAIN' ) );
+			}
+
+			if ( ! empty( $_FILES['import_file']['name'] ) ) {
+				$extension = end( explode( '.', $_FILES['import_file']['name'] ) );
+			}
+
+			if( ! isset ( $extension ) || $extension != 'json' ) {
+				wp_die( __( 'Please upload a valid .json file.', 'PLUGIN_TEXT_DOMAIN' ) );
+			}
+
+			// Retrieve the settings from the file and convert the json object to an array.
+			$settings = json_decode( file_get_contents( $import_file ), true );
+
+            if ( update_option( 'tribe_events_calendar_options', $settings ) ) {
```

Comment:
1. Before we use `$settings` we should probably do some basic validation (if we could not decode it, for instance, it will be `(bool) false` ... we might also test to ensure it is of type `array`, etc—basically, does it look the way we expect it to look?).
2. Possibly something for a future version ... might it be worth capturing the current settings and storing them in a backup option, so we can offer users the ability to quickly undo/rollback.

Reply:
#1 is done.
#2 is a fantastic idea, I love it!
Now I want to get this first version out the door. I still want to further enhance it with including options of our premium plugins, and then I will add this as well.